cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreHttpClient.java

```diff
@@ -115,35 +117,35 @@ protected boolean isAvailable() {
     }
   }
 
-  protected QueryHandle doEnableExploreStream(Id.Stream stream, String tableName,
+  protected QueryHandle doEnableExploreStream(StreamId stream, String tableName,
                                               FormatSpecification format) throws ExploreException {
     HttpResponse response = doPost(String.format(
       "namespaces/%s/data/explore/streams/%s/tables/%s/enable",
-      stream.getNamespaceId(), stream.getId(), tableName), format == null ? null : GSON.toJson(format), null);
+      stream.getNamespace(), stream.getEntityName(), tableName), format == null ? null : GSON.toJson(format), null);
```

Comment:
I am not sure. Personally don't like that we have two methods that do the same thing. ``getEntityName`` is required, since it helps even when you're not sure of the entity type. Should we do away with ``getStream``, etc?